pkg/api/customization/monitor/cluster_graph_action.go

```diff
@@ -69,7 +75,20 @@ func (h *ClusterGraphHandler) QuerySeriesAction(actionName string, action *types
 	var svcName, svcNamespace, svcPort, token string
 	if inputParser.Input.Filters["resourceType"] == "istiocluster" {
 		inputParser.Input.MetricParams["namespace"] = ".*"
-		svcName, svcNamespace, svcPort = monitorutil.IstioPrometheusEndpoint()
+
+		selector := fields.OneTermEqualSelector(systemProjectLabel, "true")
+		projects, err := h.projectsGetter.Projects(clusterName).List(metav1.ListOptions{LabelSelector: selector.String()})
+		if err != nil {
+			return err
+		}
+		if len(projects.Items) == 0 {
+			return fmt.Errorf("system project is not found by label %s=true", systemProjectLabel)
+		}
+		app, err := h.appsGetter.Apps(projects.Items[0].Name).Get(monitorutil.IstioAppName, metav1.GetOptions{})
```

Comment:
Can we use cache(`controller.Lister`) to get app here?

Reply:
updated